src/subword.rs

```diff
@@ -160,7 +158,40 @@ impl ExplicitIndexer {
             ngrams.len(),
             "ngrams contained duplicate entries."
         );
-        ExplicitIndexer { ngrams, index }
+        let bound = index.len();
+        ExplicitIndexer {
+            ngrams,
+            index,
+            bound,
+        }
+    }
+
+    /// Construct a new explicit indexer with given indices.
+    ///
+    /// The indices associated with each ngram are not stable and will only stay constant if the
+    /// indices enumerate the ngrams. In any other case, the indices are given a new index based
+    /// on the order in which they appear.
+    ///
+    /// Panics when there are duplicate ngrams.
+    pub fn new_with_indices(ngram_tuples: Vec<(String, u64)>) -> Self {
+        let mut buckets = HashMap::new();
```

Comment:
It may be strange to use the word `bucket` in the implementation here, because there is nothing in the documentation or signature that suggests (that this is restricted to) buckets.

Reply:
>     * How/where is the new `Storage` made?

That's done externally. I implemented this locally as a `finalfusion-utils` subcommand. Users can extract the `(ngram, bucket)` `Vec` from their bucket vocabulary, construct an explicit vocab and use these to construct a new storage.

>     * I think there is still a behavioral difference after the vocab conversion: before, as long as there was an n-gram within the range, we would get some (possibly random) embedding. However, after conversion, this is not true, if an n-gram did not occur in the vocab, `None` will be returned.

Yeah, but I don't think that's a bad thing. Downstream models will get some consistency wrt. oov words instead of random noise.